### DISCUSSION TITLE
Segmentation fault on Windows/scoop

### DISCUSSION THREAD

**Discussion by jackusay** [orhun/git-cliff#892] [Q&A]
install via scoop; win7

* `git cliff` shows nothing
* `git-cliff --workdir .` shows `Segmentation fault`

screenshot:
![2024-09-28 13_48_10-MINGW64__c_Users_eight_Downloads_testr_cuda_find_in_filesJ](https://github.com/user-attachments/assets/88543860-aa61-4f1b-99b1-c1dc95f3a30d)

text version:
```git
eight@eight-PC MINGW64 ~/Downloads/testr/cuda_find_in_filesJ (master)
$ scoop list
Installed apps:

Name       Version      Source Updated             Info
----       -------      ------ -------             ----
cwrsync    6.2.8        main   2023-06-16 18:32:26
gh         2.25.1       main   2023-04-01 22:45:22
git-chglog 0.15.4       main   2024-09-26 23:11:56
git-cliff  2.6.0        main   2024-09-26 21:45:55
hyperfine  1.17.0       main   2023-07-02 06:29:40
sudo       0.2014.07.18        2016-11-03 19:28:24



eight@eight-PC MINGW64 ~/Downloads/testr/cuda_find_in_filesJ (master)
$ git cliff --init

eight@eight-PC MINGW64 ~/Downloads/testr/cuda_find_in_filesJ (master)
$ git cliff

eight@eight-PC MINGW64 ~/Downloads/testr/cuda_find_in_filesJ (master)
$ git-cliff --workdir .
Segmentation fault

```

**Comment by orhun** [comment]
Hello, thanks for reporting! Can you try running `git-cliff` via `--verbose` flag (i.e. `-vv`) and see if are you getting any additional output?

**Reply by jackusay** [reply]
still get same:
```
$ git-cliff via --verbose
Segmentation fault
```

![2024-09-28 18_13_54-MINGW64__c_Users_eight_Downloads_app note_cudatext-windows-i386-1 188 0 0_py_cud](https://github.com/user-attachments/assets/362ce36d-7fc8-4a01-b948-775d998e7d5e)

**Reply by orhun** [reply]
That's weird. I don't have a windows system to reproduce this sadly.

Maybe ping @lloydjatkinson since he added the Windows package in #520

**Reply by lloydjatkinson** [reply]
Hi!

So, I have not been able to reproduce this on Windows at all, but I'm using "real" Windows, which I'll get to in a second - there's a lot of problems with this GitHub issue.

- It seems you're using MINGW64. This is what you use when you want to compile and run Linux/Unix GCC etc on Windows. `git-cliff` doesn't use this, and you don't need it to run it on Windows. Some background: https://gist.github.com/ReneNyffenegger/a8e9aa59166760c5550f993857ee437d
- Using Windows 7. This was EOL in 2020, so 4 years ago.
- Rust support for Windows 7 was dropped in 1.76, which seems to have happened around a year ago.

So in short there's really nothing that can be done here, as you need to upgrade to a supported version of Windows.

--- 

If you want to build and run Linux software on modern Windows, you should install [Windows Terminal](https://github.com/microsoft/terminal) and setup [WSL](https://learn.microsoft.com/en-us/windows/wsl/setup/environment). Additionally though, this also requires Windows 10 or Windows 11.

**Reply by orhun** [reply]
Thanks for the thorough reply!
